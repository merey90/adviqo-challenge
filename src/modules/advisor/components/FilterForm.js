import React from 'react';

function FilterForm(props) {
  const { filter, handleFormSubmit, handleInputChange } = props;
  const languages = ['all', 'english', 'kazakh'];
  return (
    <form onSubmit={handleFormSubmit}>
      <label> All: 
        <input type='radio' name='status' value='all'
          onChange={handleInputChange}
          checked={filter.status === 'all'}
        />
      </label>
      <label> Online: 
        <input type='radio' name='status' value='online'
          onChange={handleInputChange}
          checked={filter.status === 'online'}
        />
      </label>
      <label> Offline: 
        <input type='radio' name='status' value='offline'
          onChange={handleInputChange}
          checked={filter.status === 'offline'}
        />
      </label>
      <label> Language: 
        <select name='language' defaultValue={filter.language} onChange={handleInputChange}>
          {languages.map((language, index) =>
            (<option key={index} value={language}>
              {language.charAt(0).toUpperCase() + language.slice(1)}
            </option>))
          }
        </select>
      </label>
      <button type='submit'>Filter</button>
    </form>
  )
}

export default FilterForm;