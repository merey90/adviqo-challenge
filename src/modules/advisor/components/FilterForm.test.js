import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import FilterForm from './FilterForm';

const filter = {
  language: 'all',
  status: 'all',
  sortBy: 'reviews',
  sortDirection: 'desc'
};

it('should handle input change', () => {
  let inputChanged = 0;

  const handleInputChange = (e) => {
    inputChanged++;
  }

  const filterForm = shallow(
    <FilterForm
      filter={filter}
      handleInputChange={handleInputChange}
      handleFormSubmit={()=>{}}/>
  );

  filterForm.find('input[name="status"]').at(2).simulate('change');
  filterForm.find('select[name="language"]').simulate('change');

  expect(inputChanged).toEqual(2);
});

test('render FilterForm', () => {
  const component = renderer.create(
    <FilterForm
      filter={filter}
      handleInputChange={()=>{}}
      handleFormSubmit={()=>{}}/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});