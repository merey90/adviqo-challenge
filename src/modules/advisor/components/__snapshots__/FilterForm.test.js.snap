// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render FilterForm 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
     All:
    <input
      checked={true}
      name="status"
      onChange={[Function]}
      type="radio"
      value="all"
    />
  </label>
  <label>
     Online:
    <input
      checked={false}
      name="status"
      onChange={[Function]}
      type="radio"
      value="online"
    />
  </label>
  <label>
     Offline:
    <input
      checked={false}
      name="status"
      onChange={[Function]}
      type="radio"
      value="offline"
    />
  </label>
  <label>
     Language:
    <select
      defaultValue="all"
      name="language"
      onChange={[Function]}
    >
      <option
        value="all"
      >
        All
      </option>
      <option
        value="english"
      >
        English
      </option>
      <option
        value="kazakh"
      >
        Kazakh
      </option>
    </select>
  </label>
  <button
    type="submit"
  >
    Filter
  </button>
</form>
`;
